internal/datastore/crdb/stats.go

```diff
@@ -18,22 +18,33 @@ const (
 	colUniqueID   = "unique_id"
 )
 
-var (
-	queryReadUniqueID = psql.Select(colUniqueID).From(tableMetadata)
-	uniqueID          string
-)
+var queryReadUniqueID = psql.Select(colUniqueID).From(tableMetadata)
 
-func (cds *crdbDatastore) Statistics(ctx context.Context) (datastore.Stats, error) {
-	if len(uniqueID) == 0 {
+func (cds *crdbDatastore) UniqueID(ctx context.Context) (string, error) {
+	if cds.uniqueID.Load() == nil {
 		sql, args, err := queryReadUniqueID.ToSql()
 		if err != nil {
-			return datastore.Stats{}, fmt.Errorf("unable to prepare unique ID sql: %w", err)
+			return "", fmt.Errorf("unable to prepare unique ID sql: %w", err)
 		}
+
+		var uniqueID string
 		if err := cds.readPool.QueryRowFunc(ctx, func(ctx context.Context, row pgx.Row) error {
 			return row.Scan(&uniqueID)
 		}, sql, args...); err != nil {
-			return datastore.Stats{}, fmt.Errorf("unable to query unique ID: %w", err)
+			return "", fmt.Errorf("unable to query unique ID: %w", err)
 		}
+
+		cds.uniqueID.Store(&uniqueID)
```

Comment:
I don't think this matters too much here because the value converges, but this can race. Ideally we do a CAS operation.

I'd prefer to see this implemented with `sync.Once` instead of an atomic pointer. The ID does not change for the lifetime of the server.

Reply:
`sync.Once` won't work because if we get an error, we need to rerun this code the next time its called

Reply:
Fair. Can we at least use the CAS operation to store it? if it fails, it probably means some other goroutine already set it.

Another point in favor of the atomic value is it does not rely on any mutex. `sync.Once` relies on it while it's not initialized, but once it is, the overhead of the mutex disapears.

Reply:
Why even do that though? The ID is guaranteed to be the same by the datastore, so no need to compare?